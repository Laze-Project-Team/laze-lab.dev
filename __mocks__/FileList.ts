class FileList<T> {
  _files: T[];
  length: number;
  constructor(files: T[]) {
    this._files = files;
    this.length = this._files.length;
  }
  item(index: number) {
    return this._files[index];
  }
  get [Symbol.toStringTag]() {
    return 'FileList';
  }
}

// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
window.FileList = FileList;

export {};
