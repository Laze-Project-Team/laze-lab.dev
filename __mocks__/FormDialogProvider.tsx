import type { FC, ReactNode } from 'react';
import { useForm } from 'react-hook-form';

import { FormDialogContextProvider } from '@/components/ui/FormDialog/FormDialogContext';

export const FormDialogProvider: FC<{ children?: ReactNode }> = ({
  children,
}) => {
  const { control } = useForm();
  return (
    <FormDialogContextProvider
      value={{
        id: 'mock-form-dialog',
        control,
        onSubmit: async (data) => console.log(data),
      }}
    >
      {children}
    </FormDialogContextProvider>
  );
};
