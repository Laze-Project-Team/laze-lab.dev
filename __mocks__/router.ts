import type { useRouter } from 'next/router';
import { beforeEach, vi } from 'vitest';

export const mockRouter = (): void => {
  vi.mock('next/router', () => ({
    useRouter: () => {
      const router: ReturnType<typeof useRouter> = {
        asPath: '/',
        back: vi.fn(),
        basePath: '/',
        beforePopState: vi.fn(),
        events: {
          on: vi.fn(),
          off: vi.fn(),
          emit: vi.fn(),
        },
        isFallback: false,
        isLocaleDomain: false,
        isPreview: false,
        isReady: true,
        locale: 'en',
        locales: ['en'],
        prefetch: vi.fn(),
        push: vi.fn(),
        query: {},
        reload: vi.fn(),
        replace: vi.fn(),
        route: '/',
        pathname: '/',
      };

      return router;
    },
  }));
};

export const mockEachRouter = (): void => {
  beforeEach(() => {
    mockRouter();
  });
};
