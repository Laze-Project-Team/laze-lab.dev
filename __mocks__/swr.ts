import type { SWRResponse } from 'swr';

export const mockSWRResponse = <T>(data: T): SWRResponse<T> => ({
  isValidating: false,
  mutate: async () => void 0,
  data,
});
