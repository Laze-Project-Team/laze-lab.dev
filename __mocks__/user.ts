import type { User } from 'firebase/auth';

import type { userData } from '@/typings/database';

export const userMock: User | null | undefined = {
  uid: 'unique-id',
  delete: async () => void 0,
  displayName: 'Laze Project Team',
  email: 'laze@example.com',
  emailVerified: true,
  getIdToken: async () => 'token',
  getIdTokenResult: async () => ({
    authTime: new Date().toISOString(),
    expirationTime: new Date().toISOString(),
    issuedAtTime: new Date().toISOString(),
    signInProvider: null,
    signInSecondFactor: null,
    token: 'token',
    claims: {},
  }),
  isAnonymous: false,
  metadata: {},
  phoneNumber: null,
  photoURL: null,
  providerData: [],
  providerId: 'email',
  refreshToken: 'token',
  reload: async () => void 0,
  tenantId: null,
  toJSON: () => ({}),
};

export const userDataMock: userData | null | undefined = {
  name: 'Laze Project Team',
  avatarURL: 'https://laze.ddns.net/img/logo/logo.png',
  locale: 'ja',
  projects: [],
  createdAt: new Date(),
};
