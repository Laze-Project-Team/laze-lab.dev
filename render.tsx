import type {
  Queries,
  queries,
  RenderOptions,
  RenderResult,
} from '@testing-library/react';
import { render as defaultRender } from '@testing-library/react';
import type { FC, ReactNode } from 'react';

import { ColorModeProvider } from '@/components/contexts/ColorModeContext';

export type renderType<
  Q extends Queries = typeof queries,
  Container extends Element | DocumentFragment = HTMLElement,
  BaseElement extends Element | DocumentFragment = Container,
> = (
  ui: React.ReactElement,
  options?: RenderOptions<Q, Container, BaseElement>,
) => RenderResult<Q, Container, BaseElement>;

type wrapperProps = {
  children?: ReactNode;
};

type globalWrapperHOCType = (
  wrapper?: React.JSXElementConstructor<{ children: React.ReactElement }>,
) => FC<wrapperProps>;

const globalWrapperHOC: globalWrapperHOCType = (Wrapper) => {
  const globalWrapper: FC<wrapperProps> = ({ children }) => (
    <ColorModeProvider>
      {Wrapper ? (
        <Wrapper>
          <>{children}</>
        </Wrapper>
      ) : (
        children
      )}
    </ColorModeProvider>
  );

  return globalWrapper;
};

export const render: renderType = (ui, options) =>
  defaultRender(ui, {
    ...options,
    wrapper: globalWrapperHOC(options?.wrapper),
  });
