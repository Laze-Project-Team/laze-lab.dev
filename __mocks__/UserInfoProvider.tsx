import type { FC, ReactNode } from 'react';
import { vi } from 'vitest';

import { mockSWRResponse } from '/__mocks__/swr';
import { userDataMock, userMock } from '/__mocks__/user';
import { userInfoContext } from '@/components/contexts/UserInfoContext';

export const syncUserMock = vi.fn();
export const syncUserDataMock = vi.fn();

export const UserInfoProvider: FC<{ children?: ReactNode }> = ({
  children,
}) => (
  <userInfoContext.Provider
    value={{
      user: mockSWRResponse(userMock),
      userData: mockSWRResponse(userDataMock),
      syncUser: syncUserMock,
      syncUserData: syncUserDataMock,
    }}
  >
    {children}
  </userInfoContext.Provider>
);
