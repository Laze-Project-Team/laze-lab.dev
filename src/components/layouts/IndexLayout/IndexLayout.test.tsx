import '@testing-library/jest-dom';

import { screen } from '@testing-library/react';
import { expect, it } from 'vitest';

import { mockEachRouter } from '/__mocks__/router';
import { render } from '@/lib/test/render';

import { IndexLayout } from '.';

mockEachRouter();

it('should be rendered correctly', () => {
  const { container } = render(<IndexLayout />);
  expect(container.firstChild).toMatchSnapshot();
});

it('should render children', () => {
  const children = "Hey I'm children";
  render(<IndexLayout>{children}</IndexLayout>);
  expect(screen.getByText(children)).toBeInTheDocument();
});

it('should render IndexHeader', () => {
  render(<IndexLayout />);
  expect(screen.getByRole('banner')).toBeInTheDocument();
});

it('should render IndexFooter', () => {
  render(<IndexLayout />);
  expect(screen.getByRole('contentinfo')).toBeInTheDocument();
});
