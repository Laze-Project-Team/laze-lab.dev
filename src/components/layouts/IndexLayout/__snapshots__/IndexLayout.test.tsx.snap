// Vitest Snapshot v1

exports[`should be rendered correctly 1`] = `
<div
  class="css-1r26qec-PresentialIndexLayout"
>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorDefault MuiAppBar-positionStatic css-jj504k-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1joxij9-MuiToolbar-root-PresentialIndexHeader"
      >
        <a
          class="css-f6n97z-PresentialStyledLink-PresentialStyledLink-HeaderLink"
          href="/"
        >
          <span
            style="box-sizing: border-box; display: inline-block; overflow: hidden; width: initial; height: initial; background: none; opacity: 1; border: 0; margin: 0; padding: 0; position: relative; max-width: 100%;"
          >
            <span
              style="box-sizing: border-box; display: block; width: initial; height: initial; background: none; opacity: 1; border: 0; margin: 0; padding: 0; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2730%27%20height=%2730%27/%3e"
                style="display: block; max-width: 100%; width: initial; height: initial; background: none; opacity: 1; border: 0; margin: 0; padding: 0;"
              />
            </span>
            <img
              alt=""
              data-nimg="intrinsic"
              decoding="async"
              role="presentation"
              src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
              style="position: absolute; top: 0; left: 0; bottom: 0; right: 0; box-sizing: border-box; padding: 0; border: none; margin: auto; display: block; width: 0; height: 0; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
            <noscript />
          </span>
          <span
            class="css-1xs4zwo-PresentialIndexHeader"
          >
            Laze
          </span>
        </a>
        <div
          class="css-184y9ld-MuiStack-root-PresentialIndexHeader"
        >
          <button
            aria-label="switchColorMode"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-d0ntnl-MuiButtonBase-root-MuiIconButton-root-PresentialIndexHeader"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LightModeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zM2 13h2c.55 0 1-.45 1-1s-.45-1-1-1H2c-.55 0-1 .45-1 1s.45 1 1 1zm18 0h2c.55 0 1-.45 1-1s-.45-1-1-1h-2c-.55 0-1 .45-1 1s.45 1 1 1zM11 2v2c0 .55.45 1 1 1s1-.45 1-1V2c0-.55-.45-1-1-1s-1 .45-1 1zm0 18v2c0 .55.45 1 1 1s1-.45 1-1v-2c0-.55-.45-1-1-1s-1 .45-1 1zM5.99 4.58c-.39-.39-1.03-.39-1.41 0-.39.39-.39 1.03 0 1.41l1.06 1.06c.39.39 1.03.39 1.41 0s.39-1.03 0-1.41L5.99 4.58zm12.37 12.37c-.39-.39-1.03-.39-1.41 0-.39.39-.39 1.03 0 1.41l1.06 1.06c.39.39 1.03.39 1.41 0 .39-.39.39-1.03 0-1.41l-1.06-1.06zm1.06-10.96c.39-.39.39-1.03 0-1.41-.39-.39-1.03-.39-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06zM7.05 18.36c.39-.39.39-1.03 0-1.41-.39-.39-1.03-.39-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiFormControl-root css-1zjftj-MuiFormControl-root-PresentialLocaleSelectButton-PresentialIndexHeader"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              id="locale-select-label"
            >
              changeLocale
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby="locale-select-label locale-select"
                class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputSizeSmall css-jedpe8-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                id="locale-select"
                role="button"
                tabindex="0"
              >
                English
              </div>
              <input
                aria-hidden="true"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                defaultvalue="en"
                tabindex="-1"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-14lo706"
                >
                  <span>
                    changeLocale
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
    </div>
  </header>
  <main
    class="css-d8nlvz-PresentialIndexLayout"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
    />
  </main>
  <footer
    class="css-1eygvux-PresentialIndexFooter"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="css-oqt7gk-PresentialIndexFooter"
      >
        <div>
          <div
            class="css-195zqkh-PresentialIndexFooter"
          >
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; width: initial; height: initial; background: none; opacity: 1; border: 0; margin: 0; padding: 0; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; width: initial; height: initial; background: none; opacity: 1; border: 0; margin: 0; padding: 0; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2740%27%20height=%2740%27/%3e"
                  style="display: block; max-width: 100%; width: initial; height: initial; background: none; opacity: 1; border: 0; margin: 0; padding: 0;"
                />
              </span>
              <img
                alt=""
                class="css-1b8vcvq-PresentialIndexFooter"
                data-nimg="intrinsic"
                decoding="async"
                role="presentation"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0; left: 0; bottom: 0; right: 0; box-sizing: border-box; padding: 0; border: none; margin: auto; display: block; width: 0; height: 0; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
            <span
              class="css-1ouy1n3-PresentialIndexFooter"
            >
              Laze
            </span>
          </div>
          <p
            class="css-8481t7-PresentialIndexFooter"
          >
            common:motto
          </p>
          <div
            class="css-1506kd7-PresentialIndexFooter"
          >
            <a
              class="css-1qn4jyq-PresentialStyledLink-PresentialStyledLink"
              href="https://github.com/Laze-Project-Team"
              rel="noopener noreferrer"
              target="_blank"
              title="GitHub"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lpyg5v-MuiSvgIcon-root-footerIconStyle"
                data-testid="GitHubIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
                />
              </svg>
            </a>
            <a
              class="css-1qn4jyq-PresentialStyledLink-PresentialStyledLink"
              href="https://twitter.com/LazeProjectTeam"
              rel="noopener noreferrer"
              target="_blank"
              title="Twitter"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lpyg5v-MuiSvgIcon-root-footerIconStyle"
                data-testid="TwitterIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
                />
              </svg>
            </a>
            <a
              class="css-1qn4jyq-PresentialStyledLink-PresentialStyledLink"
              href="https://discord.gg/K3prMrmS7e"
              rel="noopener noreferrer"
              target="_blank"
              title="Discord"
            >
              <svg
                class="css-18rq15l-footerIconStyle"
                fill="none"
                height="55"
                role="presentation"
                viewBox="0 0 71 55"
                width="71"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  clip-path="url(#clip0)"
                >
                  <path
                    d="M60.1045 4.8978C55.5792 2.8214 50.7265 1.2916 45.6527 0.41542C45.5603 0.39851 45.468 0.440769 45.4204 0.525289C44.7963 1.6353 44.105 3.0834 43.6209 4.2216C38.1637 3.4046 32.7345 3.4046 27.3892 4.2216C26.905 3.0581 26.1886 1.6353 25.5617 0.525289C25.5141 0.443589 25.4218 0.40133 25.3294 0.41542C20.2584 1.2888 15.4057 2.8186 10.8776 4.8978C10.8384 4.9147 10.8048 4.9429 10.7825 4.9795C1.57795 18.7309 -0.943561 32.1443 0.293408 45.3914C0.299005 45.4562 0.335386 45.5182 0.385761 45.5576C6.45866 50.0174 12.3413 52.7249 18.1147 54.5195C18.2071 54.5477 18.305 54.5139 18.3638 54.4378C19.7295 52.5728 20.9469 50.6063 21.9907 48.5383C22.0523 48.4172 21.9935 48.2735 21.8676 48.2256C19.9366 47.4931 18.0979 46.6 16.3292 45.5858C16.1893 45.5041 16.1781 45.304 16.3068 45.2082C16.679 44.9293 17.0513 44.6391 17.4067 44.3461C17.471 44.2926 17.5606 44.2813 17.6362 44.3151C29.2558 49.6202 41.8354 49.6202 53.3179 44.3151C53.3935 44.2785 53.4831 44.2898 53.5502 44.3433C53.9057 44.6363 54.2779 44.9293 54.6529 45.2082C54.7816 45.304 54.7732 45.5041 54.6333 45.5858C52.8646 46.6197 51.0259 47.4931 49.0921 48.2228C48.9662 48.2707 48.9102 48.4172 48.9718 48.5383C50.038 50.6034 51.2554 52.5699 52.5959 54.435C52.6519 54.5139 52.7526 54.5477 52.845 54.5195C58.6464 52.7249 64.529 50.0174 70.6019 45.5576C70.6551 45.5182 70.6887 45.459 70.6943 45.3942C72.1747 30.0791 68.2147 16.7757 60.1968 4.9823C60.1772 4.9429 60.1437 4.9147 60.1045 4.8978ZM23.7259 37.3253C20.2276 37.3253 17.3451 34.1136 17.3451 30.1693C17.3451 26.225 20.1717 23.0133 23.7259 23.0133C27.308 23.0133 30.1626 26.2532 30.1066 30.1693C30.1066 34.1136 27.28 37.3253 23.7259 37.3253ZM47.3178 37.3253C43.8196 37.3253 40.9371 34.1136 40.9371 30.1693C40.9371 26.225 43.7636 23.0133 47.3178 23.0133C50.9 23.0133 53.7545 26.2532 53.6986 30.1693C53.6986 34.1136 50.9 37.3253 47.3178 37.3253Z"
                    fill="currentColor"
                  />
                </g>
                <defs>
                  <clippath
                    id="clip0"
                  >
                    <rect
                      fill="currentColor"
                      height="55"
                      width="71"
                    />
                  </clippath>
                </defs>
              </svg>
            </a>
            <a
              class="css-1qn4jyq-PresentialStyledLink-PresentialStyledLink"
              href="https://qiita.com/tags/laze"
              rel="noopener noreferrer"
              target="_blank"
              title="Qiita"
            >
              <svg
                class="css-18rq15l-footerIconStyle"
                role="presentation"
                version="1.2"
                viewBox="0 0 600 600"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  id="favicon"
                >
                  <g
                    id="<Group>"
                  >
                    <path
                      d="m487.6 496.5c9.3 6.2 19.1 9.9 28.6 11.4c-54.5 56.8-131.2 92.1-216.2 92.1c-165.7 0-300-134.4-300-300.1c0-165.8 134.3-300.2 300-300.2c165.7 0 300 134.4 300 300.2c0 50.9-12.7 98.8-35.1 140.8c-11.7-7.3-30.7-8.4-48.5-4.3c-11.8 2.8-22.4-13.7-29.1-22.3c-6.9-8.8-5.8-17.7-2-28.2c8.2-22.5 11.4-46.2 8.5-77c-5.1-54.6-27.8-102.7-61.2-138l-2.5-61.9c-0.8-9.7-11.1-15.3-19.2-10.4l-43.4 26.3c-27.6-12-58-17.3-89.3-14.1c-36.6 3.7-69.8 18.4-97.2 40.9l-46.2-11.2c-8.9-2.1-17.3 6.6-15.1 15.7l13.3 52.6c-22.9 39.8-34.3 88.4-29.5 139.6c11.3 120.3 104.7 168.4 212.5 157.5c53.2-5.4 97.2-16.7 129.5-51.2c8.4 18.7 24.2 29.9 42.1 41.8z"
                      fill="currentColor"
                      id="<Path>"
                    />
                    <path
                      d="m421 245.4l27.3 5.2l-0.6 5.7l-26-5.6c0.5 3.3 1 5.1 1.4 8.5c6.6 70.9-48.9 96.6-119.7 103.7c-70.8 7.2-130-6.8-136.7-77.8c-0.3-3.4-0.4-5.2-0.5-8.5l-24 11.2l-1.2-6.2l25.2-10.3c0-4.4 0.3-7.2 0.8-11.5l-27.6 2.2l-0.2-5l28.3-2.3c0.5-3.8 1.2-6.1 2.1-9.8l-32.4-3.5l0.3-6.5l33 5c13.4-50.9 57.1-90.7 112.3-96.2c55.3-5.6 104.2 23.1 126.7 70.5l33.7-11.7l1 6.4l-31.6 10.1q2.3 5.2 4.1 10.6l28.8-3.5l0.4 5.1l-28.1 3.4c1.3 4 2.3 6.6 3.2 10.8z"
                      fill="currentColor"
                      id="<Path>"
                    />
                  </g>
                </g>
              </svg>
            </a>
          </div>
          <div>
            <p
              class="css-1r885e9-PresentialIndexFooter"
            >
              common:copyright
            </p>
          </div>
        </div>
        <div
          class="css-1oifffp-PresentialIndexFooter"
        >
          <div>
            <p
              class="css-10ussqf-FooterLinkList"
            >
              footer.Guides
            </p>
            <div
              class="css-1uqyzoi-FooterLinkList"
            >
              <a
                class="css-3gu0b6-PresentialStyledLink-PresentialStyledLink-FooterLinkList"
                href="/"
              >
                footer.guides.Terms
              </a>
              <a
                class="css-3gu0b6-PresentialStyledLink-PresentialStyledLink-FooterLinkList"
                href="/guideline"
              >
                footer.guides.Guideline
              </a>
              <a
                class="css-3gu0b6-PresentialStyledLink-PresentialStyledLink-FooterLinkList"
                href="/privacy"
              >
                footer.guides.Privacy
              </a>
              <a
                class="css-3gu0b6-PresentialStyledLink-PresentialStyledLink-FooterLinkList"
                href="/license"
              >
                footer.guides.License
              </a>
            </div>
          </div>
          <div>
            <p
              class="css-10ussqf-FooterLinkList"
            >
              footer.Sitemap
            </p>
            <div
              class="css-1uqyzoi-FooterLinkList"
            >
              <a
                class="css-3gu0b6-PresentialStyledLink-PresentialStyledLink-FooterLinkList"
                href="/"
              >
                footer.sitemap.Home
              </a>
              <a
                class="css-3gu0b6-PresentialStyledLink-PresentialStyledLink-FooterLinkList"
                href="/guideline"
              >
                footer.sitemap.Editor
              </a>
              <a
                class="css-3gu0b6-PresentialStyledLink-PresentialStyledLink-FooterLinkList"
                href="/privacy"
              >
                footer.sitemap.Docs
              </a>
              <a
                class="css-3gu0b6-PresentialStyledLink-PresentialStyledLink-FooterLinkList"
                href="/license"
              >
                footer.sitemap.Compete
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
